### ISSUE TITLE
Weird issue when using TextLineReader nested in a function

### ISSUE THREAD

**Issue by gqy1005** [tensorflow/tensorflow#20640]
### System information
- WIndows 7 enterprise:
- TensorFlow installed from conda:
- TensorFlow version (1.8.0):
- Python version (3.6.5), installed with Anaconda: 
- Not using GPU
- Reproduction: Just copy the code below, it's very simple code:

### Describe the problem
I was testing the functionality of TextLineReader, a very simple try. Here is my code, just a few lines:
###### MY CODE#######
import tensorflow as tf

def input_func():
    file_queue = tf.train.string_input_producer(["D:/tmp_10_data_test.txt"], num_epochs=5)
    reader = tf.TextLineReader(skip_header_lines=1)
    return reader.read_up_to(file_queue, num_records=5)

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    sess.run(tf.local_variables_initializer())
    coord = tf.train.Coordinator()
    threads = tf.train.start_queue_runners(coord=coord, sess=sess)
    print(sess.run(input_func()))
    coord.request_stop()
    coord.join(threads)
###################
The weird part is, when I'm running this piece of code, the program keeps running without any error, any output, and it just runs forever. This issue happens when I run the code with PyCharm and Jupyter Notebook. I resolved it by moving the file_queue declaration outside the function as:
#####################
file_queue = tf.train.string_input_producer(["D:/tmp_10_data_test.txt"], num_epochs=5)
def input_func():
    reader = tf.TextLineReader(skip_header_lines=1)
    return reader.read_up_to(file_queue, num_records=5)
######################
And the program runs correctly and outputs the result as expected. But This is just weird, what happened when I declare the file queue inside the very simple function? Shouldn't it be the same regardless of where it is declared? I guess you can replace the file with your own multi-line txt file and the problem should still exist, although I didn't try it...

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by aselle** [comment]
I believe the problem is that you can't start the queue runners until you have a queue created or else it will dead lock. That is what your workaround does. You might want to take a look at using the dataset feeding mechanism which is much simpler to use and more intuitive.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by gqy1005** [comment]
Sorry for the late response. This is no more an issue for me. Thanks for
your kind explanation.

Best regards,
Qingyu

2018-08-08 3:10 GMT+08:00 Alfred Sorten Wolf <notifications@github.com>:
